app/src/main/java/nu/yona/app/api/manager/ActivityCategoryManager.java

```diff
@@ -29,4 +29,19 @@
      * @return the list of activity categories
      */
     ActivityCategories getListOfActivityCategories();
+
+
+    /**
+     * Validates the new environment by requesting the activityCategoryList api. Hence the function is added in this class.
+     *  @param url the new Environment URL entered by user
+     * @param listener the listener
+     * @return the list of activity categories
+     */
+     void validateNewEnvironment(String url,final DataLoadListener listener);
```

Comment:
Don't use ``final``, except for member variables.

Reply:
final is used the inner classes inside the function. If I don't use final, compiler throws error to do so as there is a chance it gets modified.

Reply:
In this particular case, omitting final won't lead to any compiler errors, as this is in an interface. If you use anonymous inner classes, then indeed you might need to make variables final. However, you should prevent such classes as they nearly always make your method explode beyond readability constraints.

Reply:
Lambda from your commit is implemented to get rid of final declarations.